### DISCUSSION TITLE
PowerDNS-Admin LDAP Auth

### DISCUSSION THREAD

**Discussion by NicoleLap** [ngoduykhanh/PowerDNS-Admin#1045] [Q&A] [answered]
Hello,

I'm planning to set up the LDAP Auth on our PowerDNS-Admin. Can someone confirm that I'll still be able to log in using the local DB after setting up the LDAP configuration ?

I'd like to avoid the case where I'm not able to log in anymore if I wrongly configure it... Thanks !

Regards,
Nicole

**Comment by ghost** [comment, marked as answer]
Hi @NicoleLap,

local DB authentication should continue to work as long as it's not disabled, but note that user permissions can optionally be LDAP-managed (see "Group Security" and "Advance" sections on the LDAP auth settings page), so it's good to have a clear concept of where to store what.

More specifically, in the front-end, if multiple authentication methods are enabled, a drop-down list appears on the login form that allows you to choose your authentication method in an ad-hoc manner.

In the back-end, if LDAP authentication succeeds, PowerDNS-Admin inserts the username into the local DB if it doesn't exist there already. It inserts it along with an invalid password hash (`*`) so local authentication won't work against that username - but it's possible to change the password afterwards if need be. If a record with that same username already exists in the local DB, then that record is left as-is so you'll still be able to perform local authentication against it.

**Reply by NicoleLap** [reply]
Hi !

Thanks a lot for this prompt and clear answer. To be sure, is it the same mecanism for all Auth methods ? I might be tempted to go for OpenID Auth afterall.

Regards,
Nicole
